sw/device/lib/testing/test_coverage.c

```diff
@@ -0,0 +1,84 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/uart.h"
+#include "sw/vendor/llvm_clang_rt_profile/compiler-rt/lib/profile/InstrProfiling.h"
+#include <stdint.h>
+
+/**
+ * When the linker finds a definition of this symbol, it knows to skip loading
+ * the object which contains the profiling runtime’s static initializer.
+ */
+int __llvm_profile_runtime;
+
+/**
+ * Start and end of `.init_array` section where the LLVM profiling runtime
+ * initialization function resides, defined in `flash_link.ld`.
+ */
+extern void (*_init_array_start[])(void);
+extern void (*_init_array_end[])(void);
+
+/**
+ * Computes the CRC32 of a buffer as expected by Python's `zlib.crc32()`. The
+ * implementation below is basically a simplified, i.e. byte-by-byte and without
+ * a lookup table, version of zlib's crc32. See
+ * https://github.com/madler/zlib/blob/2fa463bacfff79181df1a5270fb67cc679a53e71/crc32.c,
+ * lines 111-112 and 276-279.
+ */
+static uint32_t crc32(char *buf, size_t len) {
+  // CRC32 polynomial.
+  static const uint32_t kPoly = 0xEDB88320;
+  // Since we use a contiguous buffer, we don't need to call this function
+  // multiple times. That's why `crc` is not a parameter and is initialized to
+  // zero.
+  uint32_t crc = 0;
+  crc = ~crc;
```

Comment:
Maybe just init this to UINT32_MAX?

Reply:
(Applies to your next comment too) I implemented this as close as possible to the code in `zlib.h` to make reviewers' lives easier. Given that I didn't receive any comments regarding its correctness, I cleaned this up.